test/testdata/namer/class_and_alias.rb

```diff
@@ -1,8 +1,8 @@
 # typed: true
-A = 91
-class A # error: Redefining constant `A` as a class or module
+A = 91 # error: Cannot initialize the class `A` by constant assignment
+class A
 end
 
 class B
 end
-B = 91 # error: Redefining constant `B` as a static field
+B = 91 # error: Cannot initialize the class `B` by constant assignment
```

Comment:
This error doesn't feel particularly clear to me, especially when compared with the previous error in the `B` case.  I guess I don't mentally model `class A` as defining a constant, even though that's really what is happening in Ruby.

The extra context for the error message does clarify things somewhat, so maybe this is a nonissue.

Maybe "Cannot reassign constant `B` after it has already been declared as a class"?  I guess that gets awkward in the `A` case, but maybe you could have different messages depending on how the `Loc`s of the two definitions are ordered.  Maybe "Cannot declare `B` via constant assignment and a class definition"?

Reply:
My thinking was that most of the time when this is happening, it'll have been something like

```ruby
# -- foo.rb --

Foo = dsl_that_generates_a_class()

# -- foo.rbi --

class Foo
  def method_that_was_missing; end
end
```

The old message here would have chosen one of the two files arbitrarily (probably sorting based on FileRef ID, which is loosely alphabetic by directory order) and said that the second definition redefines the first.

Now Sorbet will always think that `Foo` is a class, and say that you can't initialize `Foo` which is a class using the `dsl_that_generates_a_class()` method.

I agree there's probably room to further improve this message. While in this test file saying "cannot reassign" reads better, in the case where the conflict comes from separate files, it's not a reassignment.

Anyways, maybe we just land the stack as is and follow up with feedback people have about the error message in practice?